libc/src/__support/FPUtil/x86_64/LongDoubleBits.h

```diff
@@ -84,13 +84,12 @@ struct FPBits<long double>
   }
 
   LIBC_INLINE constexpr void set_sign(bool signVal) {
-    bits &= ~SIGN_MASK;
-    StorageType sign1 = StorageType(signVal) << (TOTAL_LEN - 1);
-    bits |= sign1;
+    if (get_sign() != signVal)
```

Comment:
Is there any reason not to use a branchless version ? https://godbolt.org/z/4Mxso7bjG

Reply:
I don't think this works, the code "**inverts** the bit if `signVal` is set regardless of the state of `bits`" what we want is to "**set** the bit if `signVal` is set and **clear** it otherwise".

Reply:
I think this would work https://graphics.stanford.edu/~seander/bithacks.html#ConditionalSetOrClearBitsWithoutBranching but not sure if it's faster.

Reply:
Interestingly, it seems that clang is able to use `cmove` even with the first incarnation of the code (as opposed to GCC)
https://godbolt.org/z/x45jEszPe

Reply:
Ah, right. I misread the code. There should probably be documentation for the function :) Or is it elsewhere and did I miss it somehow ? Also the name is misleading (but probably in a separate change ?)

Reply:
> Ah, right. I misread the code. There should probably be documentation for the function :) Or is it elsewhere and did I miss it somehow ? Also the name is misleading (but probably in a separate change ?)

Sorry, brain fart. I got the semantics right and my implementation was incorrect. The bit hacks one looks really good, especially on GCC !